lib/install/deps.js

```diff
@@ -600,6 +600,7 @@ var earliestInstallable = exports.earliestInstallable = function (requiredBy, tr
   var binaryMatches = typeof pkg.bin === 'object' && tree.children.some(function (child) {
     if (child.removed) return false
     if (typeof child.package.bin !== 'object') return false
+    if (!pkg.bin || !child.package.bin) return false
```

Comment:
LGTM for the last change! Great :smiley_cat:

In this case, the package that has `bin: null` is [cli-size](https://github.com/IonicaBizau/cli-size/blob/master/package.json#L38) which is used by `cli-update`, so I've written the test case to reproduce the error:
https://gist.github.com/watilde/2e2533299bf2942bd719893062017307

Here is a small example to reproduce the original issue that I do some spelunking:
```json
{
  "name": "test",
  "version": "0.0.0",
  "dependencies": {
    "diable": "*",
    "cli-update": "*",
  }
}
```

Thanks!